client/db/bolt/db.go

```diff
@@ -547,7 +547,11 @@ func (db *BoltDB) LinkOrder(oid, linkedID order.OrderID) error {
 		if oBkt == nil {
 			return fmt.Errorf("LinkOrder - order %s not found", oid)
 		}
-		return oBkt.Put(linkedKey, linkedID[:])
+		linkedB := linkedID[:]
+		if linkedID.IsZero() {
+			linkedB = nil
+		}
```

Comment:
Makes sense, but I'd consider reversing the logic to skip slicing the array when not needed.
i.e.

```go
var linkedB []byte
if !linkedID.IsZero() {
	linkedB = linkedID[:]
}
```